components/layouts/AppLayout/index.js

```diff
@@ -0,0 +1,74 @@
+import { MenuOutlined } from "@ant-design/icons";
+import { Button, Drawer, Layout } from "antd";
+import { useEffect, useState } from "react";
+import AppMenu from "~/components/AppMenu";
+
+import styles from "./style.module.css";
+
+const { Content, Sider } = Layout;
+
+function AppLayout({ children, hidePrimaryMenu }) {
+  const [isMobile, setMobile] = useState();
+  const [drawerVisible, setDrawerVisible] = useState(false);
+  const [collapsed, setCollapsed] = useState();
+
+  const handleResize = () => {
+    if (window.innerWidth < 576 && !isMobile) setMobile(true);
+    else if (window.innerWidth >= 576 && isMobile) setMobile(false);
+  };
+
+  useEffect(() => {
+    if (window.innerWidth < 576) setMobile(true);
+    else {
+      setMobile(false);
+      setDrawerVisible(false);
+    }
+  }, []);
+  useEffect(() => {
+    window.addEventListener("resize", handleResize);
+  });
```

Comment:
How many times should this useEffect run?

Reply:
Only once, on page load. Why?

Reply:
Yeah, you are right, I mixed them up. Either way, this component had some bugs in that part that should be fixed